tutorial/real_proj/github-org-analyzer.md

```diff
@@ -0,0 +1,244 @@
+
+   implicit none
+   private
+
+   ! Declare the function as public to make it accessible to other modules
+   public :: print_org_repositories
+contains
+   subroutine print_org_repositories()
+      !! subroutine to print all repositories of fortran-lang organization
+      character(*), parameter :: api_url = 'https://api.github.com/orgs/fortran-lang/repos'
+      !! Stores GitHub API URL for fetching repositories
+      integer :: i
+      !! counter to traverse all the repos return by api
+      character(:), allocatable :: count
+      !! stores the string equivalent of counter, i.e variable i
+      character(:), allocatable :: value
+      !! stores the individual github repo name for each traversal
+      type(json_file) :: json
+      !! give the ability to parse the json content
+      type(response_type) :: response
+      !! stores the response from the github api
+      logical :: found
+      !!  flag for weather the current repo found or not
+
+      ! Make an HTTP GET request to the API URL and store the response in the `response` variable
+      response = request(url=api_url)
+
+      ! Checking for any errors. If an error occurs during the HTTP call, the `ok`
+      ! attribute will be set to .false., and the `err_msg` attribute will contain
+      ! the reason for the error.
+      if(.not. response%ok) then
+         print *, 'Request Fail: ', response%err_msg
+      else
+         print *, 'Fortran lang All repositories:'
+
+         ! Initialize the `json` object to parse the JSON content
+         call json%initialize()
+
+         ! Deserialize the JSON response(parsing the json)
+         call json%deserialize(response%content)
+
+         ! Traverse Repositories and Print Names
+
+         ! Counter to traverse all repos one by one
+         i = 1
+
+         ! Storing the string equivalent of i in the count variable
+         count = to_string(i)
+
+         ! Fetching the name of the 1st GitHub repository, if it exists (found is set to true)
+         call json%get('['//count//'].name', value, found)
+
+         ! Enter the loop to traverse all repositories while they exist
+         do while(found)
+
+            ! Fetch the name of the current repository (based on the `i` counter) and check if it exists
+            call json%get('['//count//'].name', value, found)
+
+            ! If the repository name exists (`found` is true), print the repository number and name
+            if (found) then
+               print *, count//'. ', value
+            end if
+
+            ! Increment the counter for the next repository
+            i = i + 1
+
+            ! Convert the updated counter to its string representation and store it in count variable
+            count = to_string(i)
+
+            ! Fetch the name of the next repository (based on the updated `i` counter) and update `found` accordingly
+            call json%get('['//count//'].name', value, found)
+
+         end do
```

Comment:
Here is a suggestion (with a single call to `json%get`) that looks easier to understand (at least to me):

```suggestion
         ! Traverse Repositories and Print Names

         ! Counter to traverse all repos one by one
         i = 0

         ! Enter the loop to traverse all repositories while they exist
         do
            ! Increment the counter for the next repository
            i = i + 1

            ! Convert the updated counter to its string representation and store it in count variable
            count = to_string(i)

            ! Fetch the name of the current repository (based on the `i` counter) and check if it exists
            call json%get('['//count//'].name', value, found)

            if(.not.found) exit

            ! If the repository name exists (`found` is true), print the repository number and name
            print *, count//'. ', value            
            
         end do
```

Reply:
Thanks, @jvdp1 now it's more readable